Exception/JWTAuthenticationException.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Exception;
+
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+/**
+ * Exception to be used during a failed JWT authentication process.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class JWTAuthenticationException extends AuthenticationException
```

Comment:
Hmm, I'm not sure about this class:

1) This extends `AuthenticationException`... and the message that will be returned to the client comes from the `getMessageKey()` method, which will be `An authentication exception occurred.` for all of these different situations. Don't we want the messages in this class to actually be shown to the client?

2) Since all these different exception "situations" are the same class, it will make it a bit harder for the end-user to potentially take different action based on what went wrong (e.g. if they want to do something different if the token is invalid versus if the token is not found). Could we use different classes?